### ISSUE TITLE
reported diagnostic span is wrong

### ISSUE THREAD

**Issue by heejaechang** [microsoft/pyright#696]
**Describe the bug**
sometimes, usually the very first time, squiggle span (diagnostic span) on an identifier is 1 off.

**To Reproduce**
paste this code in latest pyright

```python
a = pd
```

see squiggle
![image](https://user-images.githubusercontent.com/1333179/83187751-80f85480-a0e3-11ea-98eb-eda8037aff72.png)

**Expected behavior**
span should cover whole identifier "pd"

**VS Code extension or command-line**
extension

**Comment by erictraut** [comment]
@heejaechang I figured out the repro that was eluding me. It needs to be pasted at the very end of the file with no carriage return at the end of the line. Problem is now fixed.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.40, which I just published.
